### ISSUE TITLE
xpra server leaks memory

### ISSUE THREAD

**Issue by totaam** [Xpra-org/xpra#2730]
*Issue migrated from trac ticket # 2730*

**component:** server | **priority:** major | **resolution:** fixed

#### 2020-04-16 08:43:19: mafjmafj created the issue

___
> 
> xpra v3.0.7-25629
> 
> ```
> 
> Fedora Core 31 5.5.11-200.fc31.x86_64 #1 SMP Mon Mar 23 17:32:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
> 
> ```
> 
> I started happening after recent dnf update
> 
> Similar to #1662
> 
> ```
> 
> [mj@mj-opti7040 ~]$ xpra info :122 | grep memory
> threads.memory.children.idrss=0
> threads.memory.children.inblock=107368
> threads.memory.children.isrss=0
> threads.memory.children.ixrss=0
> threads.memory.children.majflt=383
> threads.memory.children.maxrss=5342608
> threads.memory.children.minflt=116275
> threads.memory.children.msgrcv=0
> threads.memory.children.msgsnd=0
> threads.memory.children.nivcsw=278
> threads.memory.children.nsignals=0
> threads.memory.children.nswap=0
> threads.memory.children.nvcsw=2460
> threads.memory.children.oublock=56
> threads.memory.children.stime=0
> threads.memory.children.utime=2
> threads.memory.server.idrss=0
> threads.memory.server.inblock=135360
> threads.memory.server.isrss=0
> threads.memory.server.ixrss=0
> threads.memory.server.majflt=1710
> threads.memory.server.maxrss=5794408
> threads.memory.server.minflt=3353250
> threads.memory.server.msgrcv=0
> threads.memory.server.msgsnd=0
> threads.memory.server.nivcsw=27229
> threads.memory.server.nsignals=0
> threads.memory.server.nswap=0
> threads.memory.server.nvcsw=2581653
> threads.memory.server.oublock=16
> threads.memory.server.stime=90
> threads.memory.server.utime=1065
> total-memory=8221163520
> 
> ```
> 
> The only apps in use: terminator, meld and boostnote
> Session was running for several days.
> Session reattached dozens of times.
> 
> ```
> 
> xpra stop :122
> ```
>  made the xpra process go away and memory freed
> ```
> 2383977 mj        20   0 7459300   5.1g  27072 S   0.0  66.4  19:12.37 xpra
> ```
> 
>

**Comment by allo-** [comment]
Should this bug be fixed in 3.0.13? I definitely have a memory leak on the server with the Debian package "3.0.13+dfsg1-1~bpo10+1".

```
threads.memory.children.idrss=0
threads.memory.children.inblock=0
threads.memory.children.isrss=0
threads.memory.children.ixrss=0
threads.memory.children.majflt=0
threads.memory.children.maxrss=119376
threads.memory.children.minflt=40142
threads.memory.children.msgrcv=0
threads.memory.children.msgsnd=0
threads.memory.children.nivcsw=288
threads.memory.children.nsignals=0
threads.memory.children.nswap=0
threads.memory.children.nvcsw=1349
threads.memory.children.oublock=344
threads.memory.children.stime=0
threads.memory.children.utime=0
threads.memory.server.idrss=0
threads.memory.server.inblock=155160
threads.memory.server.isrss=0
threads.memory.server.ixrss=0
threads.memory.server.majflt=2143
threads.memory.server.maxrss=4268712
threads.memory.server.minflt=36217898
threads.memory.server.msgrcv=0
threads.memory.server.msgsnd=0
threads.memory.server.nivcsw=7315982
threads.memory.server.nsignals=0
threads.memory.server.nswap=0
threads.memory.server.nvcsw=25232972
threads.memory.server.oublock=176
threads.memory.server.stime=651
threads.memory.server.utime=9286
total-memory=8365920256
```

**Comment by totaam** [comment]
@allo- the backport to v3.0.x was in 449836d4f4c05f04d128c3e48b8755dd29a8c226 and was included in 3.0.13
(but the Debian packages are not supported - use the ones from xpra.org to report bugs)

**Comment by allo-** [comment]
I am currently on 4.4.4 and it seems to be (slowly) leaking.

I use the xpra.org Debian packages. I may be a bit behind as the repo seems to have problems with an expired GPG key at the moment, though.
